### PR TITLE
Fix testing farm related bugs after refactoring

### PR DESCRIPTION
**Pull request by lbarcziova**
Should fix https://sentry.io/share/issue/66c4594b0d314f1c88c385c96c3792e6/ and https://sentry.io/share/issue/751ffcb263254f108bae9ec85e0b2213/

### CODE REVIEW

File: packit_service/worker/tasks.py [hunk 1]
```diff
@@ -226,7 +227,9 @@ def run_testing_farm_results_handler(
         package_config=load_package_config(package_config),
         job_config=load_job_config(job_config),
         data=EventData.from_event_dict(event),
-        tests=event.get("tests"),
+        tests=[TestResult(**test) for test in event.get("tests")]
+        if event.get("tests")
+        else [],
```
Comment (lachmanfrantisek) [inline packit_service/worker/tasks.py:232]:
```suggestion
        tests=[TestResult(**test) for test in event.get("tests", [])]
```

File: packit_service/worker/handlers/github_handlers.py [hunk 2]
```diff
@@ -468,6 +468,7 @@ def __init__(
         data: EventData,
         chroot: str,
     ):
+        data.trigger = TheJobTriggerType.pull_request
```
Comment (lachmanfrantisek) [inline packit_service/worker/handlers/github_handlers.py:471]: How do we know it's a pull request? What about tests for branch push?

We should probably update `triggers` to support all three (release, PR, commit).

Comment (lbarcziova) [reply packit_service/worker/handlers/github_handlers.py:471]: OK, this [line](https://github.com/packit-service/packit-service/blob/17ccef68ecd5cc12282f6555ede80e18f5f80a7f/packit_service/worker/handlers/github_handlers.py#L462) confused me. 

Then forwarding the build model id should be better (and using then `build.job_trigger.get_trigger_object()`)

Comment (lachmanfrantisek) [reply packit_service/worker/handlers/github_handlers.py:471]: > OK, this [line](https://github.com/packit-service/packit-service/blob/17ccef68ecd5cc12282f6555ede80e18f5f80a7f/packit_service/worker/handlers/github_handlers.py#L462) confused me.

So did me.

File: packit_service/worker/handlers/github_handlers.py [hunk 3]
```diff
@@ -459,19 +463,33 @@ class GithubTestingFarmHandler(JobHandler):
...
             package_config=package_config, job_config=job_config, data=data,
         )
         self.chroot = chroot
+        self.build_id = build_id
+        self._db_trigger: Optional[AbstractTriggerDbType] = None
+
+    @property
+    def db_trigger(self) -> Optional[AbstractTriggerDbType]:
+        if not self._db_trigger:
+            build = CoprBuildModel.get_by_id(self.build_id)
```
Comment (lachmanfrantisek) [inline packit_service/worker/handlers/github_handlers.py:490]: Just to be sure. What about returning `None` when no build is found?

Comment (lbarcziova) [reply packit_service/worker/handlers/github_handlers.py:490]: I was thinking about this, but in CoprBuildEndHandler if the build is not found, the testing farm handler task is not called.

Comment (lachmanfrantisek) [reply packit_service/worker/handlers/github_handlers.py:490]: ok

**Pull request merged**